src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1627,6 +1660,7 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
       out.print_raw   (cmd);
       out.print_raw_cr("\" ...");
 
+      VMErrorThreadToNativeFromVM ttnfv(JavaThread::current_or_null());
```

Comment:
Surely the current thread need not be a JavaThread here.

Reply:
Make sense. I haven't seen report_and_die() is called by NonJavaThread, but I agree we should cover that case.  it will be no-op for NonJavaThread because safepoint synchronization only checks JavaThreads.

Reply:
I would move the RAII object out of this scope into the enclosing scope. Do it at the start of `if OnError[0] != NULL`. There is no need to repeatedly do this for every command.

(See my earlier comment, maybe we don't even need an RAII but a simple transition, done once, would be sufficient).